src/k8s/pkg/k8s/setup/kubeconfigs.go

```diff
@@ -58,11 +67,87 @@ func InitKubeconfigs(ctx context.Context, state *state.State, ca *cert.CertKeyPa
 			return fmt.Errorf("could not generate auth token for %s: %w", config.username, err)
 		}
 
-		err = utils.GenerateKubeconfig(token, ca.CertPem, config.path)
+		err = GenerateKubeconfig(token, ca.CertPem, config.path, overwrites)
 		if err != nil {
 			return fmt.Errorf("failed to generate kubeconfig for %s: %w", config.username, err)
 		}
 	}
 
 	return nil
 }
+
+// GenerateX509Kubeconfig creates a kubeconfig file with the given x509 certificate and key data.
+func GenerateX509Kubeconfig(keyPem, certPem, caCertPem []byte, path string, overwrites *KubeConfigOverwriteOpts) error {
```

Comment:
Moved this here because of cyclic dependencies.
Also, that methods are only used in this file, so it makes sense to have it contained in one place.